n3fit/src/n3fit/msr.py

```diff
@@ -80,16 +80,18 @@ def generate_msr_model_and_grid(
     x_divided = xDivide()(x_original)
 
     # 3. Prepare the pdf for integration by dividing by x
-    pdf_integrand = Lambda(op.op_multiply, name="pdf_integrand")([x_divided, pdf_xgrid_integration])
+    pdf_integrand = Lambda(
+        lambda x_pdf: op.batchit(x_pdf[0], batch_dimension=-1) * x_pdf[1], name="pdf_integrand"
```

Comment:
This would be much nicer of course, but as far as I know Keras forces you to use one argument functions. Just tested again `mylayer = Lambda(lambda x, y: x + y)` and whether I call it simply as `mylayer(2, 3)` or `mylayer([2, 3])` it will complain that the `y` argument is missing.